src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -1,74 +1,69 @@
 import React, { useState, useEffect, useRef } from 'react'
 import { View, FlatList, ActivityIndicator } from 'react-native'
 import { createStackNavigator } from 'react-navigation-stack'
-import { isNull, isEmpty } from 'lodash'
+import { isEmpty } from 'lodash'
 import * as AnalyticsService from '~/analyticsService'
-import { ForceRerenderOnNavigation } from '~/navigationService'
 import * as DatabaseService from '~/databaseService'
-import { sortDiagnosesByMostRecentCreation } from '~/mixins/diagnose'
-import { getURL } from '~/mixins/storage'
-import { renderDiagnoses } from './renderUtilities'
+import { renderDiagnose } from './renderUtilities'
 import Background from '~/components/Background'
 import NoDiagnoses from './components/NoDiagnoses'
 import Header from './components/Header'
 import DetailedDiagnose from '../DetailedDiagnose'
 import { OFFSET_THRESHOLD_TO_SNAP_FLATLIST } from './constants'
 import styles from './styles'
 
+const buildDiagnose = (doc) => {
+  const diagnose = doc.data({ serverTimestamps: 'estimate' })
+  diagnose.thumbnail = diagnose.imageReferences[0]
+  return diagnose
+}
+
 const MyDiagnoses = () => {
   const flatListRef = useRef()
-  const [diagnoses, setDiagnoses] = useState(null)
+  const diagnoses = useRef([])
```

Comment:
Why are you using a reference here instead of a state? If you need this value for rendering, then you _have to_ use `useState`. References don't cause re-renders of the component. All the values you use for rendering must be immutable.